src/protocols/vnc/vnc.c

```diff
@@ -385,6 +385,12 @@ void* guac_vnc_client_thread(void* data) {
     }
 #endif
 
+    /* Disable remote console (Server input) */
+    if (settings->disable_server_input) {
+        rfbBool success = PermitServerInput(rfb_client, 1);
+        guac_client_log(client, GUAC_LOG_INFO, "Disabling server input %s.", success ? "succeeded" : "failed");
```

Comment:
There is admittedly not much documentation on the return value of `PermitServerInput()`, but reading through the libvncclient code, the value doesn't appear to indicate whether disabling input succeeded:

```c
/*
 * UltraVNC Server Input Disable
 * Apparently, the remote client can *prevent* the local user from interacting with the display
 * I would think this is extremely helpful when used in a HelpDesk situation
 */
rfbBool PermitServerInput(rfbClient* client, int enabled)
{
    rfbSetServerInputMsg msg;

    if (!SupportsClient2Server(client, rfbSetServerInput)) return TRUE;
    /* enabled==1, then server input from local keyboard is disabled */
    msg.type = rfbSetServerInput;
    msg.status = (enabled ? 1 : 0);
    msg.pad = 0;
    return  (WriteToRFBServer(client, (char *)&msg, sz_rfbSetServerInputMsg) ? TRUE : FALSE);
}
```

(https://github.com/LibVNC/libvncserver/blob/c82b3abb369983cd737190331b897814a580a980/libvncclient/rfbproto.c#L1502-L1517)

Given the above, it looks like:

* If the VNC server supports the relevant message, the return value indicates only if the message was successfully sent, not whether the server honored the message.
* If the VNC server _doesn't_ support the relevant message, `PermitServerInput()` pretends to succeed.

The silent success of `PermitServerInput()` makes this log message less useful. If you want to keep the message, I suggest moving it to the `GUAC_LOG_DEBUG` level and rephrasing to capture only what we know.

Reply:
As stated in your comment it appears that the status is only related to if the sending of the message succeeds. I have adjusted the message for this:

`guac_client_log(client, GUAC_LOG_DEBUG, "%s send request to disable server input", success ? "Successfully" : "Failed to");`

In addition to this, after retesting the code locally PermitServerInput does not seem to work with some vnc clients even though the vnc clients do support the functionality.
After going through the libvncserver code I found the following:

```	/** Which messages are supported by the server
	 * This is a *guess* for most servers.
	 * (If we can even detect the type of server)
	 *
	 * If the server supports the "rfbEncodingSupportedMessages"
	 * then this will be updated when the encoding is received to
	 * accurately reflect the servers capabilities.
	 */
	rfbSupportedMessages supportedMessages;
```

It would appear that libvncserver guesses if the vnc server supports disabling the remote input and if it guesses the server does not, then PermitServerInput  will not send the request even if the client supports it.

For this reason I have changed the code back to use WriteToRFBServer, in my opinion it is better to always send the request.